engine/collection/proposal/engine.go

```diff
@@ -254,6 +257,14 @@ func (e *Engine) BroadcastProposalWithDelay(header *flow.Header, delay time.Dura
 		return fmt.Errorf("could not get cluster members: %w", err)
 	}
 
+	// adjust the delay based on time spent on block proposal construction
+	elapsed := time.Since(start)
```

Comment:
I don't think it's necessary to keep these, the difference is trivial, not worth the complexity.

A nice to have thing is to see these modified delays as metrics tho.

Reply:
do you suggest I add it as a second parameter to ClusterBlockProposed method or define a new method?

Reply:
Yes, that works. It's better to just pass the data needs to be collected to `ClusterBlockProposed`, and we can decide how to send the metrics in the metrics module. 

The `Histogram` type of metrics would be a good fit for it.

Reply:
We already have pretty detailed tracing metrics for the [collection block builder](https://github.com/onflow/flow-go/blob/03c34e747f83c205e5f6393053dd254d399908fa/module/trace/constants.go#L23-L29), so we already have insight into block building time there. Adding a delay metric would be mostly duplicating that with less granularity. I think it would be a better investment to add similar tracing to the consensus block builder instead.

Reply:
ok I just removed the unnecessary changes, and won't add the metric as it is already captured somewhere else.

Reply:
Add tracing for consensus block builder in this PR: https://github.com/onflow/flow-go/pull/42